File: scripts/delete_admin_user.py
#!/usr/bin/env python3
"""
Delete a temporary admin user by email from admin_platform.admin_users.
Usage (PowerShell):
  python scripts/delete_admin_user.py --correo admin.temp@ventas.local
Env overrides (optional): PGHOST, PGPORT, PGDATABASE, PGUSER, PGPASSWORD
"""
import os
import sys
import argparse
import psycopg2


def get_conn():
    host = os.getenv('PGHOST', 'localhost')
    port = int(os.getenv('PGPORT', '5432'))
    db = os.getenv('PGDATABASE', 'sistemas_ventas')
    user = os.getenv('PGUSER', 'admin')
    pwd = os.getenv('PGPASSWORD', 'Ventas2025')
    return psycopg2.connect(host=host, port=port, dbname=db, user=user, password=pwd)


def main():
    ap = argparse.ArgumentParser()
    ap.add_argument('--correo', required=True)
    args = ap.parse_args()
    sql = 'DELETE FROM admin_platform.admin_users WHERE correo = %s RETURNING id, correo;'

    try:
        with get_conn() as conn:
            with conn.cursor() as cur:
                cur.execute(sql, (args.correo,))
                row = cur.fetchone()
                if row:
                    print('Usuario eliminado:', row)
                else:
                    print('No se encontró usuario con ese correo')
        return 0
    except Exception as e:
        print('Error eliminando usuario:', e, file=sys.stderr)
        return 1


if __name__ == '__main__':
    sys.exit(main())

File: src/scripts/deleteAdminUser.js
import dotenv from 'dotenv';
dotenv.config();

import { query } from '../config/db.js';

function parseArgs(argv) {
  const out = {};
  for (let i = 2; i < argv.length; i++) {
    const a = argv[i];
    if (a.startsWith('--')) {
      const [k, v] = a.replace(/^--/, '').split('=');
      if (typeof v === 'undefined') out[k] = true; else out[k] = v;
    }
  }
  return out;
}

async function main() {
  const args = parseArgs(process.argv);
  const correo = args.correo || 'admin.temp@ventas.local';
  const { rows } = await query('DELETE FROM admin_platform.admin_users WHERE correo=$1 RETURNING id, correo', [correo]);
  if (rows.length) console.log('Usuario eliminado:', rows[0]);
  else console.log('No se encontró usuario con ese correo');
}

main().catch((e) => {
  console.error('Error delete admin:', e);
  process.exit(1);
});
